index.js

```diff
@@ -24,6 +25,32 @@ const modalBody = ReadElement('.modal-layout');
 const wrapper = document.querySelector('.wrapper');
 const wrapperVisible = wrapper.getAttribute('data-visible');
 
+const formObject = {
+  name: '',
+  email: '',
+  message: '',
+};
+
+function formFunc() {
+  formObject.name = inputName.value;
+  formObject.email = inputEmail.value;
+  formObject.message = inputMessage.value;
+  const stringForm = JSON.stringify(formObject);
+  localStorage.setItem('form', stringForm);
+}
+
+const localFormObject = JSON.parse(localStorage.getItem('form'));
+
+function formPopulate() {
+  if (localFormObject) {
+    inputName.value = localFormObject.name;
+    inputEmail.value = localFormObject.email;
+    inputMessage.value = localFormObject.message;
+  }
+}
+
+formPopulate();
```

Comment:
- [ ]  When the user loads the page, the code checks if there are any saved data in the local storage and if so the input fields are pre-filled with the data. YOu actually did this, but it will be better to use the `onload` event listener to prevent any rendering error.

e.g.
```
window.onload = () => {
  let savedFormData = localStorage.getItem('contactFormInput');
  savedFormData = JSON.parse(savedFormData);
  // Check if the form data object is found on localStorage
  if (savedFormData) {
  // populate inputs values if data was found
  // ex: nameInput.value = savedFormData.name
  }
```

Reply:
This feature has been corrected.
Thanks for the feedback